### PR TITLE
feat(presets)!: rename `config:base` to `config:recommended`

### LINKED ISSUE #12024: Rename `config:base` -> `config:recommended`
### What would you like Renovate to be able to do?

Quote from @rarkins [^quote]

> BTW I was already thinking that `config:base` is somewhat misnamed. Even though we try not to keep it too opinionated, it still is. I had been thinking to rename it to `config:recommended` soon, so people understand from the name it's subjective/opinionated.

### If you have any ideas on how this should be implemented, please tell us here.

- [ ] Change `config:base` to `config:recommended` in code.
- [ ] Change `config:base` to `config:recommended` in our presets.
- [ ] Write migration code.
- [ ] Change references to `config:base` -> `config:recommended` in documentation.

### Is this a feature you are interested in implementing yourself?

No

[^quote]: [Source of quote](https://github.com/renovatebot/renovate/issues/11987#issuecomment-934211664)

### PR DESCRIPTION
**Pull request by viceice**
## Changes
rename with migration

## Context
- closes #12024

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/config/presets/common.ts [hunk 1]
```diff
@@ -13,11 +13,11 @@ export const removedPresets: Record<string, string | null> = {
   ':switchToGradleLite': null,
   ':unpublishSafe': 'npm:unpublishSafe',
   'config:application': 'config:js-app',
-  'config:base-js': 'config:base',
+  'config:recommended-js': 'config:recommended',
```
Comment (rarkins) [inline lib/config/presets/common.ts:16]:
```suggestion
  'config:base': 'config:recommended',
  'config:base-js': 'config:recommended',
```

File: docs/usage/config-presets.md [hunk 2]
```diff
@@ -111,27 +111,27 @@ It mostly uses Renovate config defaults but adds a few smart customizations such
 
 ## How to Use Preset Configs
 
-By default, Renovate App's onboarding PR suggests the `["config:base]"` preset.
-If you're self hosting, and want to use the `config:base` preset, then you must add `"onboardingConfig": { "extends": ["config:base"] }` to your bot's config.
+By default, Renovate App's onboarding PR suggests the `["config:recommended]"` preset.
+If you're self hosting, and want to use the `config:recommended` preset, then you must add `"onboardingConfig": { "extends": ["config:recommended"] }` to your bot's config.
```
Comment (rarkins) [inline docs/usage/config-presets.md:115]: Maybe it's time to make this the onboardingConfig too?

**Review by rarkins — CHANGES_REQUESTED** [review]

File: docs/usage/config-presets.md (hunk 2 above)
Comment (viceice) [reply docs/usage/config-presets.md:115]: yes, did I miss that anywhere?

File: lib/config/presets/internal/config.ts [hunk 3]
```diff
@@ -3,7 +3,15 @@ import type { Preset } from '../types';
...
+  'js-app': {
+    description: 'Default configuration for webapps.',
+    extends: ['config:recommended', ':pinAllExceptPeerDependencies'],
+  },
+  'js-lib': {
+    description: 'Default configuration for libraries.',
+    extends: ['config:recommended', ':pinOnlyDevDependencies'],
+  },
+  recommended: {
     description: 'Default base configuration for all languages.',
```
Comment (HonkingGoose) [inline lib/config/presets/internal/config.ts:15]:
```suggestion
    description: 'Recommended configuration for most users. It does not matter what programming language you use.',
```

Is this the correct place to edit the `description` that we show in our docs?

I think we should explain to our users that the `config:recommended` preset is the "basic" preset that the maintainers of Renovate "recommend" to most users, no matter what programming language they use.

By saying `most users` we leave the door open to recommend the `config:best-practice` preset as the ultimate best thing to do. 😉

**Comment by viceice** [comment]
We'll retarget when we prepare v36, until ten it's easier to rebase onto main.

**Comment by HonkingGoose** [comment]
@viceice I think we're working on `v36` now, so it's probably a good time to re-target this PR to the `v36` branch. 😉

**Comment by rarkins** [comment]
Does this handle the scenario where a self-hosted user has configured an onboardingConfig containing `config:base`? Ideally we should raise the actual onboarding PR with `config:recommended` not `config:base`

File: docs/development/creating-editing-renovate-presets.md [hunk 4]
```diff
@@ -27,7 +27,7 @@ We have multiple kinds of `group:` presets, with different rules.
 ##### Rules for `group:*` presets
```
Comment (HonkingGoose) [inline docs/development/creating-editing-renovate-presets.md:27]: There are still mentions of `config:base` in these files:

- [ ] `docs/usage/reading-list.md`
- [ ] `docs/usage/key-concepts.md`

File: docs/usage/config-presets.md (hunk 2 above)
Comment (HonkingGoose) [reply docs/usage/config-presets.md:115]: Is `config:recommended` now the onboarding configuration?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]

**Review by rarkins — APPROVED** [review]

**Comment by dpprdan** [comment]
@viceice Shouldn't 
https://github.com/renovatebot/renovate/blob/f19b3001847be370f50d1ed5253e726b55391477/lib/config/presets/internal/config.ts#L10 
also become `config:recommended`?

**Comment by HonkingGoose** [comment]
@dpprdan Good catch, see issue:

- #23326

**Pull request merged**